### DISCUSSION TITLE
Hardware in loop testing of radar with carla

### DISCUSSION THREAD

**Discussion by markusmat97** [carla-simulator/carla#5527] [Q&A]
Does Carla supports hardware in the loop(HIL) testing specifically for radar?Do anyone have any ideas regarding this?Which is the best simulator for HIL testing?

**Comment by chenhengwei1999** [comment]
I had similar thoughts to you. I want to test different parameters of the millimeter wave radar for possible problems in traffic scenarios, but the carla built-in radar does not specify hardware parameter settings, such as signal waveform(FMCW, pulse wave, etc.), number of transmitting and receiving antennas, the EM band used, etc.

But I found in the carla tutorial (https://carla.readthedocs.io/en/latest/ecosys_ansys/) a link that seems to say that ANSYS designed a custom radar and tested it in the carla environment.

I'm currently exploring the available methods, so feel free to explore together!

**Reply by markusmat97** [reply]
My idea is create a scenario from real driving data(from publicly available datasets) and run it in carla and integrate this model for HIL testing.Do you think this will work?or instead of carla is it possible to do the same with IPG carmaker?
